src/Analyser/Scope.php

```diff
@@ -790,7 +790,10 @@ private function resolveType(Expr $node): Type
 			$leftType = $this->getType($left);
 			$rightType = $this->getType($right);
 
-			if ($leftType instanceof StringType && $rightType instanceof StringType) {
+			if (
+				(new StringType())->isSuperTypeOf($leftType)->yes()
+				&& (new StringType())->isSuperTypeOf($rightType)->yes()
```

Comment:
Please don't create `new StringType()` each time. You can reuse the same instance for both conditions and for the `return` statement.